docs/components/compact.md

```diff
@@ -9,37 +9,160 @@ menu: components
+external_labels: {cluster="us1", replica="1", receive="true", environment="production"}
+external_labels: {cluster="us1", replica="1", receive="true", environment="staging"}
+```
+
+and set `--deduplication.replica-label="replica"`, compactor will assume those as:
+
+```
+external_labels: {cluster="eu1", receive="true", environment="production"} (2 streams, resulted in one)
+external_labels: {cluster="us1", receive="true", environment="production"}
+external_labels: {cluster="us1", receive="true", environment="staging"}
+```
+
+On next compaction multiple streams' blocks will be compacted into one.
+
+## Enforcing Retention of Data
 
-## Downsampling, Resolution and Retention
+By default, there is NO retention set for object storage data. This means that you store data for unlimited time, which is a valid and recommended way of running Thanos.
 
-Resolution - distance between data points on your graphs. E.g.
+You can set retention by different resolutions using `--retention.resolution-raw` `--retention.resolution-5m` and `--retention.resolution-1h` flag. Not setting
+them or setting to `0s` means no retention.
 
-- raw - the same as scrape interval at the moment of data ingestion
-- 5m - data point is every 5 minutes
-- 1h - data point is every 1h
+**NOTE:** ⚠ ️Retention is applied at the end of compaction loop right after Compation and Downsamping loops. If those are failing, data will be never deleted.
 
-Keep in mind, that the initial goal of downsampling is not saving disk space (Read further for elaboration on storage space consumption). The goal of downsampling is providing an opportunity to get fast results for range queries of big time intervals like months or years. In other words, if you set `--retention.resolution-raw` less then `--retention.resolution-5m` and `--retention.resolution-1h` - you might run into a problem of not being able to "zoom in" to your historical data.
+## Downsampling
+
+Downsampling is a process of rewriting series' to reduce overall resolution of the samples without loosing accuracy over longer time ranges.
+
+To learn more see [video from KubeCon 2019](https://youtu.be/qQN0N14HXPM?t=714)
+
+### ⚠ ️Downsampling: Note About Resolution and Retention ⚠️
+
+Resolution is a distance between data points on your graphs. E.g.
+
+* `raw` - the same as scrape interval at the moment of data ingestion
+* `5 minutes` - data point is every 5 minutes
+* `1 hour` - data point is every 1h
+
+Keep in mind, that the initial goal of downsampling is not saving disk or object storage space. In fact, downsampling doesn't save you __any__ space but instead,
+it adds 2 more blocks for each raw block which are only slightly smaller or relatively similar size to raw block. This is done by internal downsampling implementation
+which to be mathematically correct holds various aggregations. This means that downsampling can increase the size of your storage a bit (~3x), if you choose to store all resolutions (recommended and by default).
```

Comment:
Are you sure about 3x? That seems higher than I would expect. Do we have measurements for this?

Reply:
It's technically 2 more "streams" of blocks so it's up to 3x

Reply:
It’s been a while for me to have looked at this part but I thought we don’t duplicate the index, but just samples in chunks. Or is it one block per resolution?

Reply:
We duplicate. It's a separate block - not tied to any specific block for now. Definitely to improve (: (issue to be added for it)